modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -50,3 +74,254 @@ function manatime_feature_timesheets_cron() {
   }
 }
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function manatime_feature_timesheets_ctools_plugin_directory($owner, $plugin_type) {
+  if ($owner === 'ctools' && $plugin_type === 'content_types') {
+    return 'plugins/content_types';
+  }
+  if ($owner === 'ctools' && $plugin_type === 'relationships') {
+    return 'plugins/relationships';
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function manatime_feature_timesheets_theme() {
+  return array(
+    'timesheet-view' => array(
+      'variables' => array(
+        'id_timesheet' => '',
+        'records' => array(),
+        'add_row_link' => '',
+      ),
+      'template' => 'timesheet-view',
+      'path' => drupal_get_path('module', 'manatime_feature_timesheets') . '/themes',
+    ),
+  );
+}
+
+/**
+ * Add a new row in the current timesheet.
+ */
+function manatime_feature_timesheets_add_row_form($form, &$form_state) {
+  $timesheet = new Timesheet($form_state['build_info']['args'][0]);
+  // Get the list of options to the dropdown project.
+  $options_project = timesheet_get_projects_options();
+  // Selected default option to the dropdown project.
+  $selected = isset($form_state['values']['project']) ? $form_state['values']['project'] : key($options_project);
+  $form['project'] = array(
+    '#type' => 'select',
+    '#title' => t('Project'),
+    '#options' => $options_project,
+    '#default_value' => $selected,
+    '#ajax' => array(
+      'callback' => 'refresh_tasks_callback',
+      'wrapper' => 'field_task',
+    ),
+  );
+  $form['task'] = array(
+    '#type' => 'select',
+    '#title' => t('Task'),
+    // Create the section to update in the DOM.
+    '#prefix' => '<div id="field_task">',
+    '#suffix' => '</div>',
+    '#options' => timesheet_get_tasks_options($selected, $timesheet),
+    '#default_value' => isset($form_state['values']['task']) ? $form_state['values']['task'] : '',
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save'),
+  );
+  return $form;
+}
+
+/**
+ * Remove a row in the current timesheet.
+ */
+function manatime_feature_timesheets_remove_row_form($form, &$form_state) {
+  $form['alert'] = array(
+    '#type' => 'item',
+    '#title' => t('Are you sure to delete records?'),
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Remove'),
+  );
+  return $form;
+}
+
+/**
+ * Helper function to make a link add row.
+ */
+function manatime_feature_timesheet_make_link_add_row($nid) {
+  ctools_include('modal');
+  ctools_modal_add_js();
+  return l(t('Add row'), 'timesheet/add-new-form/' . $nid . '/nojs', array('attributes' => array('class' => 'ctools-use-modal')));
+}
+
+/**
+ * Helper function to populate the project dropdown.
+ */
+function timesheet_get_projects_options() {
+  $list = array();
+  $query = new EntityFieldQuery();
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'projects')
```

Comment:
We need to get only active projects. Please add a condition (I think there's an active field in the project)